Feat(eos_validate_state): Added the validation for STUN client configurations (#3898)
aristanetworks/ansible-avd: ansible_collections/arista/avd/roles/eos_validate_state/ANTA-Preview.md

- (New) AvdTestAPIHttpsSSL (No Ansible tags, use the new `skipped_tests` variable instead)
  - VerifyAPIHttpsSSL: Validate eAPI HTTPS SSL profile status.

- (New) AvdTestStun (No Ansible tags, use the new `skipped_tests` variable instead)

>   - AvdTestStun: Validate STUN clients source address and port.

Reviewer: ```suggestion
  - VerifyStunClient: Validate STUN clients source address and port.
```

>   - AvdTestStun: Validate STUN clients source address and port.

Reviewer: Can we add more details about the test here? Since it's related to WAN devices, we should mention it somehow.

>   - VerifyStunClient: Validate the STUN client is configured with the specified IPv4 source address and port exclusively for WAN scenarios. Additionally, validate the public IP and port if provided. Please add the test in a custom ANTA catalog to validate public IP and port.

Reviewer 2: 1 lets remove the part where we suggest custom ANTA catatlog for public IP and port. AVD does not offer this capability to be derived from the structured config

2. update using the suggestion I made in the test docstring. What the test is really checking is that there is an entry in the STUN client translations for the given local interfaces in each path-group.

>   - VerifyStunClient: Validate the STUN client is configured with the specified IPv4 source address and port. Validate the public IP and port if provided. For now AVD use source port as static 4500 and do not validate public IP and port. Please add the test in custom anta catalogs to validate public ip and port.

Reviewer 2: ```suggestion
  - VerifyStunClient: Validate the STUN client is configured with the specified IPv4 source address and port. Validate the public IP and port if provided. For now AVD sets the source port to 4500 and does not validate public IP and port. Please add the test in a custom ANTA catalog to validate public IP and port.
```

Reviewer 2: Remove the source port 4500 for now. Add stuff about the fact that we do STUN check only for WAN case for now (because we look for path-groups)
